app/src/main/java/com/example/umlandowallet/Start.kt

```diff
@@ -170,14 +177,40 @@ object LDKBroadcaster: BroadcasterInterface.BroadcasterInterfaceInterface {
 }
 
 fun initializeNetworkGraph(genesisBlockHash: String, logger: Logger) {
+    val service = Service.create()
+
     val f = File(Global.homeDir+ "/" + Global.prefixNetworkGraph);
     if (f.exists()) {
         println("loading network graph...")
         val serializedGraph = File(Global.homeDir+ "/" + Global.prefixNetworkGraph).readBytes()
         val readResult = NetworkGraph.read(serializedGraph, logger)
         if (readResult is Result_NetworkGraphDecodeErrorZ.Result_NetworkGraphDecodeErrorZ_OK) {
             Global.router = readResult.res
-            println("loaded network graph ok")
+
+            val rapidGossipSync = RapidGossipSync.of(Global.router);
+            var latestGraphSnapshot = ByteArray(0)
+            var latestRapidSyncTimestamp =  ""
+
+            val myOption = Global.router!!._last_rapid_gossip_sync_timestamp
+
+            if (myOption is Option_u32Z.None) {
+                println("myOption None: $myOption")
+                val tsLong = System.currentTimeMillis() / 1000
+                latestRapidSyncTimestamp = tsLong.toString()
```

Comment:
the default should be 0. I would actually simply declare `latestRapidSyncTimestamp = 0` outside this if clause, and then only check the `myOption is Option_u32Z.Some` variant